addons/fleet/models/fleet_vehicle.py

```diff
@@ -0,0 +1,549 @@
+        if (operator == '=' and value is True) or (operator in ('<>', '!=') and value is False):
+            search_operator = 'in'
+        else:
+            search_operator = 'not in'
+        today = fields.Date.context_today(self)
+        datetime_today = fields.Datetime.from_string(today)
+        limit_date = fields.Date.to_string(datetime_today + relativedelta(days=+15))
+        renewal_contracts = self.env['fleet.vehicle.log.contract'].search([('expiration_date', '>', today),
+                                                                         ('expiration_date', '<', limit_date), ('state', 'in', ('open', 'toclose'))])
+        renewal_contract_vehicle = [contract.vehicle_id.id for contract in renewal_contracts]
+        return [('id', search_operator, renewal_contract_vehicle)]
+
+
+class FleetVehicleOdometer(models.Model):
+
+    _name = 'fleet.vehicle.odometer'
+    _description = 'Odometer log for a vehicle'
+    _order = 'date desc'
+
+    name = fields.Char(compute='_compute_vehicle_log_name', store=True)
+    date = fields.Date(default=fields.Date.context_today)
+    value = fields.Float('Odometer Value', group_operator="max")
+    vehicle_id = fields.Many2one('fleet.vehicle', string='Vehicle', required=True)
+    unit = fields.Selection(related='vehicle_id.odometer_unit', readonly=True)
+
+    @api.one
+    @api.depends('vehicle_id', 'date')
+    def _compute_vehicle_log_name(self):
+        name = self.vehicle_id and self.vehicle_id.name or ''
+        self.name = self.date and name + ' / ' + self.date or name
+
+
+class FleetVehicleLogFuel(models.Model):
+
+    _name = 'fleet.vehicle.log.fuel'
+    _description = 'Fuel log for vehicles'
+    _inherits = {'fleet.vehicle.cost': 'cost_id'}
+
+    @api.model
+    def default_get(self, fields):
+        res = super(FleetVehicleLogFuel, self).default_get(fields)
+        service_type = self.env.ref('fleet.type_service_refueling').id
+        res.update({'cost_type': 'fuel', 'cost_subtype_id': service_type})
+        return res
+
+    liter = fields.Float()
+    price_per_liter = fields.Float()
+    purchaser_id = fields.Many2one('res.partner', string='Purchaser',
+                                   domain="['|', ('customer', '=', True), ('employee', '=', True)]")
+    invoice_reference = fields.Char()
+    vendor_id = fields.Many2one('res.partner', string='Supplier', domain="[('supplier', '=', True)]")
+    notes = fields.Text()
+    cost_id = fields.Many2one('fleet.vehicle.cost', string='Cost', required=True, ondelete='cascade')
+    cost_amount = fields.Float(related='cost_id.amount', string='Amount', store=True)
+    date = fields.Date(default=fields.Date.context_today)
+    # we need to keep this field as a related with store=True because the graph view doesn't support (1) to address
+    # fields from inherited table and (2) fields that aren't stored in database
+
+    @api.onchange('vehicle_id')
+    def on_change_vehicle(self):
```

Comment:
The onchange methods are invoked on a pseudo-record that contains the values present in the form and it doesn't return record value now(it deals with value in cache), so one isn't going to get any benefit if we make it public.